lib/fog/azurerm/models/storage/files.rb

```diff
@@ -0,0 +1,31 @@
+require 'fog/core/collection'
+require 'fog/azurerm/models/storage/file'
+
+module Fog
+  module Storage
+    class AzureRM
+      # This class is giving implementation of listing blobs.
+      class Files < Fog::Collection
+        model Fog::Storage::AzureRM::File
+        attribute :directory
+
+        def all(options = { metadata: true })
+          files = []
+          service.list_blobs(directory, options).each do |blob|
+            hash = File.parse blob
+            hash['directory'] = directory
+            files << hash
+          end
+          load files
+        end
+
+        def get(directory, name)
+          file = File.new(service: service)
```

Comment:
The collection's get method is to get a local object which is a reference to the cloud blob object, it may not exist on the server. So creating a local object can save calls to uploading a real blob. With this local object, I can operate against it with actual REST API calls, such as: `blob.exists?`, `blob.save` and `blob.delete`, etc.